sys/arm64/arm64/trap.c

```diff
@@ -297,13 +297,17 @@ print_registers(struct trapframe *frame)
 {
 	u_int reg;
 
+	/*
+	 * TODO: We use uint64_t to be compatible with aarch64, but should
+	 * use the macros to print the full capability.
+	 */
 	for (reg = 0; reg < nitems(frame->tf_x); reg++) {
 		printf(" %sx%d: %16lx\n", (reg < 10) ? " " : "", reg,
-		    frame->tf_x[reg]);
+		    (uint64_t)frame->tf_x[reg]);
 	}
-	printf("  sp: %16lx\n", frame->tf_sp);
-	printf("  lr: %16lx\n", frame->tf_lr);
-	printf(" elr: %16lx\n", frame->tf_elr);
+	printf("  sp: %16lx\n", (uint64_t)frame->tf_sp);
```

Comment:
```suggestion
#if __has_feature(capabilities)
	printf("  sp: " _CHERI_PRINTF_CAP_FMT "\n",
	     _CHERI_PRINTF_CAP_ARG((void * __capability)frame->tf_sp));
#else
	printf("  sp: %16lx\n", (uint64_t)frame->tf_sp);
#endif
```
Or the RISC-V version:
```c
#if __has_feature(capabilities)
/* This cannot use _CHERI_PRINTF_CAP_ARG due to the cast for purecap. */
#ifdef __CHERI_PURE_CAPABILITY__
#define	PRINT_REG_ARG(value)	((void * __capability)(value))
#else
#define	PRINT_REG_ARG(value)	(&(value))
#endif
#define PRINT_REG(name, value)					\
	printf(name " = " _CHERI_PRINTF_CAP_FMT "\n",		\
	    PRINT_REG_ARG(value));
#define PRINT_REG_N(name, n, array)				\
	printf(name "[%d] = " _CHERI_PRINTF_CAP_FMT "\n", n,	\
	    PRINT_REG_ARG((array)[n]));
#else
#define PRINT_REG(name, value)	printf(name " = 0x%016lx\n", value)
#define PRINT_REG_N(name, n, array)	\
	printf(name "[%d] = 0x%016lx\n", n, (array)[n])
#endif
```